### PR TITLE
feat(lnurl-pay): handle fetch invoice errors

### LINKED ISSUE #761: Handle LNURL pay callback errors
**Describe the bug**
Currently we do not handle errors of the LNURL-pay callback

**To Reproduce**
1. pay an LNURL-pay with a certain minSendable
2. try to send a payment below the minimum
3. the error response from the callback is not handled

**Expected behavior**
Show an error alert to the user

related code: https://github.com/getAlby/lightning-browser-extension/blob/master/src/app/screens/LNURLPay.tsx#L127

### PR DESCRIPTION
**Pull request by reneaaron**
#### Link this PR to an issue
Fixes #761 

#### Type of change (Remove other not matching type)

- Bug fix (non-breaking change which fixes an issue)

#### Describe the changes you have made in this PR -

Error handling for fetching the invoice via lnurl-pay.

#### Screenshots of the changes (If any) -

image

#### How Has This Been Tested?
 - [x] Try to pay to a Lightning Address with below minimum allowed amount

#### Checklist:

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

File: src/app/screens/LNURLPay.tsx [hunk 1]
```diff
@@ -106,13 +106,20 @@ function LNURLPay(props: Props) {
...
+        const request = await axios.get<LNURLPaymentInfo>(details.callback, {
           params,
-        }
-      );
-      const { pr: paymentRequest } = paymentInfo;
+        });
+        paymentInfo = request.data;
+        paymentRequest = paymentInfo.pr;
+      } catch (e) {
+        alert("Payment aborted: Could not fetch invoice.");
```
Comment (bumi) [inline src/app/screens/LNURLPay.tsx:120]: The LNURL server can also return an "error" in the response. 
Right now we do not handle this one either. 

```
{"status":"ERROR", "reason":"error details..."}
```

I guess checking for the status "error" and then doing an alert(reason) would be good?

**Review by bumi — CHANGES_REQUESTED** [review]
thanks, let's directly also add a check for an lnurl error response

File: src/app/screens/LNURLPay.tsx (hunk 1 above)
Comment (reneaaron) [reply src/app/screens/LNURLPay.tsx]: Thanks for the hint, I just added the reason to the alert displayed to the user (if available). 

image

Comment (bumi) [reply src/app/screens/LNURLPay.tsx]: can you check the spec. I think that the service can also respond with a HTTP 200 BUT have an error field.

File: src/app/screens/LNURLPay.tsx [hunk 2]
```diff
@@ -106,13 +107,36 @@ function LNURLPay(props: Props) {
...
+        response = await axios.get<LNURLPaymentInfo | LNURLPaymentInfoError>(
+          details.callback,
+          {
+            params,
+            // https://github.com/fiatjaf/lnurl-rfc/blob/luds/01.md#http-status-codes-and-content-type
+            validateStatus: () => true,
+          }
+        );
+
+        const isSuccessResponse = function (
```
Comment (reneaaron) [inline src/app/screens/LNURLPay.tsx:123]: Do you know the reason the spec decided against HTTP status codes? Doesn't that make the implementation / tooling more complicated for all services consuming those APIs? 

If you can think of a more elegant way to solve that, please let me know!

Comment (bumi) [reply src/app/screens/LNURLPay.tsx:141]: nice, implementation. 

No idea why status codes should not be respected. honestly I am still OK with respecting them. If a server for example returns a 500 as a valid response, then imo the server is so broken, that I do not respect the response :) 

I think in other places we do assume a success response. 
but this PR looks small and clean to me.  👍

**Review by bumi — APPROVED** [review]

**Pull request merged**